helix-term/src/commands.rs

```diff
@@ -1908,21 +1904,22 @@ fn search_impl(
     // the text before the current cursor position for matches, but by slicing
     // it out, we need to add it back to the position of the selection.
     let mut offset = 0;
+    let doc = doc!(editor).text().slice(..);
 
     // use find_at to find the next match after the cursor, loop around the end
     // Careful, `Regex` uses `bytes` as offsets, not character indices!
     let mut mat = match direction {
```

Comment:
I've got a new panic here-ish with the latest changes to the branch:

<details><summary>Backtrace</summary>

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Byte index out of bounds: byte index 108636, Rope/RopeSlice byte length 79016', /home/michael/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ropey-1.6.1/src/slice.rs:324:41
stack backtrace:
   0: rust_begin_unwind
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/std/src/panicking.rs:578:5
   1: core::panicking::panic_fmt
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/panicking.rs:67:14
   2: core::result::unwrap_failed
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/result.rs:1687:5
   3: core::result::Result<T,E>::unwrap
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/result.rs:1089:23
   4: ropey::slice::RopeSlice::byte_to_char
             at /home/michael/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ropey-1.6.1/src/slice.rs:324:9
   5: helix_term::commands::search_impl
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/commands.rs:1940:21
   6: helix_term::commands::search_next_or_prev_impl
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/commands.rs:2042:17
   7: helix_term::commands::search_prev
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/commands.rs:2064:5
   8: helix_term::commands::MappableCommand::execute
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/commands.rs:209:41
   9: helix_term::ui::editor::EditorView::handle_keymap_event::{{closure}}
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/ui/editor.rs:837:13
  10: helix_term::ui::editor::EditorView::handle_keymap_event
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/ui/editor.rs:863:17
  11: helix_term::ui::editor::EditorView::command_mode
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/ui/editor.rs:969:17
  12: <helix_term::ui::editor::EditorView as helix_term::compositor::Component>::handle_event
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/ui/editor.rs:1334:33
  13: helix_term::compositor::Compositor::handle_event
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/compositor.rs:152:19
  14: helix_term::application::Application::handle_terminal_events::{{closure}}
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/application.rs:671:22
  15: helix_term::application::Application::event_loop_until_idle::{{closure}}
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/application.rs:325:55
  16: helix_term::application::Application::event_loop::{{closure}}
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/application.rs:299:57
  17: helix_term::application::Application::run::{{closure}}
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/application.rs:1227:38
  18: hx::main_impl::{{closure}}
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/main.rs:161:53
  19: tokio::runtime::park::CachedParkThread::block_on::{{closure}}
             at /home/michael/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/park.rs:282:63
  20: tokio::runtime::coop::with_budget
             at /home/michael/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/coop.rs:107:5
  21: tokio::runtime::coop::budget
             at /home/michael/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/coop.rs:73:5
  22: tokio::runtime::park::CachedParkThread::block_on
             at /home/michael/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/park.rs:282:31
  23: tokio::runtime::context::blocking::BlockingRegionGuard::block_on
             at /home/michael/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/context/blocking.rs:66:9
  24: tokio::runtime::scheduler::multi_thread::MultiThread::block_on::{{closure}}
             at /home/michael/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/scheduler/multi_thread/mod.rs:87:13
  25: tokio::runtime::context::runtime::enter_runtime
             at /home/michael/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/context/runtime.rs:65:16
  26: tokio::runtime::scheduler::multi_thread::MultiThread::block_on
             at /home/michael/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/scheduler/multi_thread/mod.rs:86:9
  27: tokio::runtime::runtime::Runtime::block_on
             at /home/michael/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.35.1/src/runtime/runtime.rs:350:45
  28: hx::main_impl
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/main.rs:163:5
  29: hx::main
             at /nix/persist/home/michael/src/helix/hx/helix-term/src/main.rs:37:21
  30: core::ops::function::FnOnce::call_once
             at /rustc/90c541806f23a127002de5b4038be731ba1458ca/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

</details>

Reproduction steps:

* Open up this file: https://github.com/elixir-mint/mint/blob/321c8303ed626452c8b010b605950ec62e23e7e2/lib/mint/http2.ex
* Search for `handle_data`
* Hit `N` to search for that pattern backwards

Reply:
fixed, thanks! Was a dump little oversight in the helix code (I fixed a bug in the way the old code called regex and forgot to remove workaround we had for our buggy code :D)